dev-portal/src/services/api-catalog.js

```diff
@@ -1,144 +1,183 @@
-      subscriptions = data
-
-      return subscriptions
+let catalogPromiseCache // WARNING: Don't touch this. Should only be used by updateCatalogAndApisList.
+
+/**
+ * Return the API with the provided apiId. Can also provide the special strings "FIRST" or "ANY" to get the first API returned. Can select the api returned as a side-effect.
+ * 
+ * @param {String} apiId   An apiId or the special strings 'FIRST' or 'ANY'. 'FIRST' and 'ANY' both return the first api encountered.
+ * @param {Boolean} [selectIt=false]   If true, sets the found API as the current 'selected' API.
+ */
+export function getApi(apiId, selectIt = false) {
+  return updateCatalogAndApisList()
+    .then(() => {
+      let thisApi
+  
+      if (store.apiList.length) {
+        if (apiId === 'ANY' || apiId === 'FIRST') {
+          thisApi = store.apiList[0]
+        }
+
+        else {
+          thisApi = store.apiList.find(api => api.id === apiId)
+        }
+      }
+
+      if (selectIt) store.api = thisApi
+
+      return thisApi
     })
-  })
 }
 
-export function clearSubscriptions() {
-  subscriptions = null
-}
+/* Subscription Utils */
 
-export function isSubscribed(usagePlanId) {
-  return !!getSubscribedUsagePlan(usagePlanId)
-}
+/**
+ * Fetch and update subscriptions store. Uses caching to determine if it should actually fetch or return the stored result.
+ * 
+ * @param {Boolean} [bustCache=false]   Ignore the cache and re-make the network call. Defaults to false.
+ */
+export function updateSubscriptions(bustCache = false) {
+  let subscriptionsOrPromise = store.subscriptions.length ? store.subscriptions : subscriptionsPromiseCache
+  if (!bustCache && subscriptionsOrPromise) return Promise.resolve(subscriptionsOrPromise)
 
-// export function getUsagePlanApiStages(usagePlanId) {
-//   const subscribedUsagePlan = getSubscribedUsagePlan(usagePlanId)
-//
-//   return (subscribedUsagePlan && subscribedUsagePlan.apiStages) || []
-// }
+  return subscriptionsPromiseCache = apiGatewayClient()
+    .then(apiGatewayClient => apiGatewayClient.get('/subscriptions', {}, {}, {}))
+    .then(({ data }) => store.subscriptions = data)
+}
+let subscriptionsPromiseCache // WARNING: Don't touch this. Should only be used by updateCatalogAndApisList.
 
 export function getSubscribedUsagePlan(usagePlanId) {
-  const subscribedUsagePlan = subscriptions && subscriptions.find && subscriptions.find(s => s.id === usagePlanId)
-  return subscribedUsagePlan
+  return store.subscriptions.find(sub => sub.id === usagePlanId)
 }
 
-export function addSubscription(usagePlanId) {
-    return getApiGatewayClient().then(apiGatewayClient => {
-        return apiGatewayClient.put('/subscriptions/' + usagePlanId, {}, {}).then((result) => {
-          window.location.reload()
-      })
-    })
+export function subscribe(usagePlanId) {
+  let localClient
+
+  return apiGatewayClient()
+    .then(apiGatewayClient => (localClient = apiGatewayClient))
```

Comment:
Same as above.

Reply:
This one doesn't need to be, but it's not harmful. That said, I'll change it -- localClient is only used once, so we've got no reason to store it anymore. (It used to have multiple invocations.)